Extension/src/Debugger/nativeAttach.ts

```diff
@@ -164,30 +165,95 @@ export class PsProcessParser {
+        let stdout: string = "";
+        let stderr: string = "";
+
+        if (process) {
+            let cancellationTokenListener: vscode.Disposable | undefined; // eslint-disable-line prefer-const
+            // Handle timeout
+            const seconds: number = 30;
+            const processTimeout: NodeJS.Timeout = setTimeout(() => {
+                process.removeAllListeners();
+                if (cancellationTokenListener) {
+                    cancellationTokenListener.dispose();
+                }
 
-            if (error) {
-                reject(error);
+                try {
+                    process.kill();
+                } catch (e) {
+                    // Failed to kill process.
+                }
+                reject(new Error(localize("timeout.processList.spawn", '"{0}" timed out after {1} seconds.', command, seconds)));
+                return;
+            }, seconds * 1000);
+
+            // Handle cancellation
+            cancellationTokenListener = token?.onCancellationRequested(() => {
+                clearTimeout(processTimeout);
+                process.removeAllListeners();
+
+                try {
+                    process.kill();
+                } catch (e) {
+                    // Failed to kill process.
+                }
+                reject(new Error(localize("cancel.processList.spawn", '"{0}" canceled.', command)));
                 return;
+            });
+
+            const cleanUpCallbacks = () => {
+                clearTimeout(processTimeout);
+                process.removeAllListeners();
+                if (cancellationTokenListener) {
+                    cancellationTokenListener.dispose();
+                }
+            };
+
+            // Handle data streams
+            if (process.stdout) {
+                process.stdout.on('data', (data: string) => {
+                    stdout += data.toString();
+                });
             }
 
-            if (stderr && stderr.length > 0) {
-                if (stderr.indexOf('screen size is bogus') >= 0) {
-                    // ignore this error silently; see https://github.com/microsoft/vscode/issues/75932
-                    // see similar fix for the Node - Debug (Legacy) Extension at https://github.com/microsoft/vscode-node-debug/commit/5298920
-                } else {
-                    reject(new Error(stderr));
+            if (process.stderr) {
+                process.stderr.on('data', (data: string) => {
+                    stderr += data.toString();
+                });
+            }
+
+            // Handle process exit
+            process.on('close', (code: number) => {
+                cleanUpCallbacks();
+                if (code !== 0) {
+                    reject(new Error(localize("error.processList.spawn", '"{0}" exited with code: "{1}".', command, code)));
                     return;
                 }
```

Comment:
Should we check stderr first since that would be easier to understand the failure?

Reply:
Or maybe include both?